src/vcon/MarketGovernance.sol

```diff
@@ -0,0 +1,653 @@
+
+    /// ---------- Permissionless User PCV Allocation Methods ----------
+
+    /// @notice a user can get slashed up to their full VCON stake for entering
+    /// a venue that takes a loss.
+    /// @param amountVcon to stake on destination
+    /// @param destination address to accrue rewards to, and send funds to
+    function stake(
+        uint256 amountVcon,
+        address destination
+    ) external globalLock(1) whenNotPaused {
+        IPCVOracle oracle = pcvOracle();
+        require(
+            oracle.isVenue(destination),
+            "MarketGovernance: invalid destination"
+        );
+
+        _accrue(destination); /// update share price in the destination so the user gets in at the current share price
+
+        /// vconToShares will always return correctly as accrue will set venueLastRecordedVconSharePrice
+        /// to the correct share price from 0 if uninitialized
+        uint256 userShareAmount = vconToShares(destination, amountVcon);
+
+        /// user updates
+        venueUserShares[destination][msg.sender] += userShareAmount;
+
+        /// venue updates
+        venueTotalShares[destination] += userShareAmount;
+
+        /// check and an interaction with a trusted contract
+        vcon().safeTransferFrom(msg.sender, address(this), amountVcon); /// transfer VCON in
+
+        emit Staked(destination, msg.sender, amountVcon);
+    }
+
+    /// @notice unstake VCON and transfer corresponding VCON to another venue
+    /// @param shareAmount the amount of shares to unstake
+    /// @param source address to accrue rewards to, and pull funds from
+    /// @param vconRecipient address to receive the VCON
+    /// @dev both source and destination are checked twice,
+    /// the first time in the market governance contract,
+    /// the second time in the PCV Router contract.
+    function unstake(
+        uint256 shareAmount,
+        address source,
+        address vconRecipient
+    ) external globalLock(1) whenNotPaused {
+        /// ---------- Checks ----------
+        IPCVOracle oracle = pcvOracle();
+
+        require(oracle.isVenue(source), "MarketGovernance: invalid source");
+        require(
+            venueUserShares[source][msg.sender] >= shareAmount,
+            "MarketGovernance: invalid share amount"
+        );
+
+        address denomination = IPCVDepositV2(source).balanceReportedIn();
+        address destination = underlyingTokenToHoldingDeposit[denomination];
+        require(
+            destination != address(0),
+            "MarketGovernance: invalid destination"
+        );
+
+        /// ---------- Effects ----------
+
+        _accrue(source); /// update profitPerVCON in the source so the user gets paid out at the current share price
+
+        /// figure out how balanced the system is before withdraw
+
+        /// amount of PCV to withdraw is the amount vcon * venue balance / total vcon staked on venue
+        uint256 amountPcv = getProRataPCVAmounts(source, shareAmount);
```

Comment:
technically this is a pro rata amount of _balance_ (number of tokens) and not an amount of _pcv_ (in USD), i'd rename the function to be clear